src/Orchard.Environment.Extensions/Features/FeatureInfo.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.DotNet.ProjectModel.FileSystemGlobbing;
+using Microsoft.Extensions.FileProviders;
+using System;
+using System.Linq;
+
+namespace Orchard.Environment.Extensions.Features
+{
+    public class FeatureInfo : IFeatureInfo
+    {
+        private string _id;
+        private string _name;
+        private double _priority;
+        private IExtensionInfo _extension;
+        private string[] _dependencies;
+
+        public FeatureInfo(
+            string id,
+            string name,
+            double priority,
+            IExtensionInfo extension,
+            string[] dependencies)
+        {
+            _id = id;
+            _name = name;
+            _priority = priority;
+            _extension = extension;
+            _dependencies = dependencies;
+        }
+
+        public string Id { get { return _id; } }
+        public string Name { get { return _name; } }
+        public double Priority { get { return _priority; } }
+        public IExtensionInfo Extension { get { return _extension; } }
+        public string[] Dependencies { get { return _dependencies; } }
+
+        public bool DependencyOn(IFeatureInfo subject)
+        {
+            return this.Dependencies.Any(x =>
```

Comment:
DependsOn? Also, if it just accesses members from the interface, then make it an extension method on the interface, it can be in the same file as the interface.